src/plone/app/mosaic/browser/static/js/mosaic.actions.js

```diff
@@ -251,6 +253,66 @@ define([
             }
         });
 
+        // Register link action
+        $.mosaic.registerAction('plonelink', {
```

Comment:
This is that ugly, because LinkModal is not really designed to be used outside TinyMCE pattern, and Mosaic is not ready to use TinyMCE pattern.